google-cloud-storage/src/test/java/com/google/cloud/storage/jqwik/StorageArbitraries.java

```diff
@@ -122,6 +140,65 @@ public static final class Buckets {
 
     private Buckets() {}
 
+    public Arbitrary<Bucket.Lifecycle.Rule> rule() {
+      Arbitrary<Bucket.Lifecycle.Rule.Action> actions =
+          Arbitraries.of(
+              Bucket.Lifecycle.Rule.Action.newBuilder().setType("delete").build(),
+              Bucket.Lifecycle.Rule.Action.newBuilder()
+                  .setType("setStorageClass")
+                  .setStorageClass("COLDLINE")
+                  .build());
+      Arbitrary<Boolean> conditionIsLive = bool();
+      Arbitrary<Integer> conditionAgeDays = Arbitraries.integers().between(0, 100);
+      Arbitrary<Integer> conditionNumberOfNewVersions = Arbitraries.integers().between(0, 10);
+      Arbitrary<Date> conditionCreatedBeforeTime = date();
+      Arbitrary<Integer> conditionDaysSinceNoncurrentTime = Arbitraries.integers().between(0, 10);
+      Arbitrary<Date> conditionNoncurrentTime = date();
+      Arbitrary<Integer> conditionDaysSinceCustomTime = Arbitraries.integers().between(0, 10);
+      Arbitrary<Date> conditionCustomTime = date();
+      ListArbitrary<String> storageClassMatches =
+          Arbitraries.of("COLDLINE", "NEARLINE", "STANDARD", "ARCHIVE").list().uniqueElements();
+
+      return Combinators.combine(
+              actions,
+              Combinators.combine(
+                      conditionIsLive,
+                      conditionAgeDays,
+                      conditionNumberOfNewVersions,
+                      conditionCreatedBeforeTime,
+                      conditionDaysSinceNoncurrentTime,
+                      conditionNoncurrentTime,
+                      conditionDaysSinceCustomTime,
+                      conditionCustomTime)
+                  .as(Tuple::of),
+              storageClassMatches)
+          .as(
+              (a, ct, s) ->
+                  Bucket.Lifecycle.Rule.newBuilder()
+                      .setAction(a)
+                      .setCondition(
+                          Bucket.Lifecycle.Rule.Condition.newBuilder()
+                              .setIsLive(ct.get1())
+                              .setAgeDays(ct.get2())
+                              .setNumNewerVersions(ct.get3())
+                              .setCreatedBefore(ct.get4())
+                              .setDaysSinceNoncurrentTime(ct.get5())
+                              .setNoncurrentTimeBefore(ct.get6())
+                              .setDaysSinceCustomTime(ct.get7())
+                              .setCustomTimeBefore(ct.get8())
+                              .addAllMatchesStorageClass(s)
+                              .build())
+                      .build());
+    }
+
+    public Arbitrary<Bucket.Lifecycle> lifecycle() {
+      return rule()
+          .list()
+          .ofMinSize(0)
+          .ofMaxSize(100)
```

Comment:
Is there an actual limit on the gcs side for how many rules can be present? If so it might make sense to use a similar value here.

Reply:
I thought it was 100 but i can't seem to find a limit. I can remove the upperbound but 100 feels okay to me. What do you think?

Reply:
100's probably fine in that case. We probably won't want to somehow end up slowing down tests because the generator needs to construct like 700k of these things.